Mathlib/LinearAlgebra/Dual.lean

```diff
@@ -604,6 +604,20 @@ def evalEquiv : M ≃ₗ[R] Dual R (Dual R M) :=
     (evalEquiv R M).symm.dualMap = Dual.eval R (Dual R M) := by
   ext; simp
 
+lemma dualMap_dualMap_comp {M' : Type*} [AddCommGroup M'] [Module R M'] {f : M →ₗ[R] M'} :
+    f.dualMap.dualMap = Dual.eval R M' ∘ₗ f ∘ₗ (evalEquiv R M).symm := by
+  ext x g
+  simp only [dualMap_apply, coe_comp, LinearEquiv.coe_coe, Function.comp_apply, Dual.eval_apply]
+  rw [← apply_evalEquiv_symm_apply, dualMap_apply]
+
+lemma dualMap_dualMap_eq_iff {M' : Type*} [AddCommGroup M'] [Module R M'] {f g : M →ₗ[R] M'}
+    (h : Injective (Dual.eval R M')) : f.dualMap.dualMap = g.dualMap.dualMap ↔ f = g := by
```

Comment:
How about we add a convenience version for when `M'` is also reflexive. So we could call this:
```suggestion
lemma dualMap_dualMap_eq_iff_of_injective
    {M' : Type*} [AddCommGroup M'] [Module R M'] {f g : M →ₗ[R] M'}
    (h : Injective (Dual.eval R M')) :
    f.dualMap.dualMap = g.dualMap.dualMap ↔ f = g := by
```
and then below add:
```lean
@[simp]
lemma dualMap_dualMap_eq_iff
    {M' : Type*} [AddCommGroup M'] [Module R M'] [IsReflexive R M'] {f g : M →ₗ[R] M'} :
    f.dualMap.dualMap = g.dualMap.dualMap ↔ f = g :=
  dualMap_dualMap_eq_iff_of_injective _ _ (bijective_dual_eval R M').injective
```